torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2428,7 +2432,8 @@ c10::intrusive_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
       device,
       rank,
       opType,
-      seqCollective_,
+      isP2P ? seqP2P_ : seqCollective_,
+      isP2P,
```

Comment:
@fduwjj @c-p-i-o do you think we can consolidate the `bool isP2P = isP2POp(opType)` logic below with this change? It may not be so trivial though. See this comment: https://github.com/pytorch/pytorch/pull/134578/files#r1764430474

Reply:
Sure. I can try to address this in a subsequent change.